webapp/loggedin/modules/data/validationReport/validationReportView.js

```diff
@@ -0,0 +1,129 @@
+import './validationReportView.scss'
+
+import React from 'react'
+import { connect } from 'react-redux'
+
+import * as R from 'ramda'
+
+import TableView from '@webapp/loggedin/tableViews/tableView'
+import { useI18n, useOnUpdate } from '@webapp/commonComponents/hooks'
+
+import * as Survey from '@core/survey/survey'
+import * as Record from '@core/record/record'
+import * as NodeDef from '@core/survey/nodeDef'
+import * as Authorizer from '@core/auth/authorizer'
+
+import { appModuleUri, dataModules } from '@webapp/loggedin/appModules'
+
+import * as AppState from '@webapp/app/appState'
+import * as SurveyState from '@webapp/survey/surveyState'
+import * as RecordsState from '@webapp/loggedin/modules/data/records/recordsState'
+
+import { reloadListItems } from '@webapp/loggedin/tableViews/actions'
+
+import ValidationFieldMessages from '@webapp/commonComponents/validationFieldMessages'
+
+const validationReportModule = 'validationReport'
+
+const ValidationReportRowHeader = ({ nodeDefKeys }) => {
+  const i18n = useI18n()
+
+  return (
+    <>
+      <div>#</div>
+      <div>{i18n.t('common.path')}</div>
+      <div>{i18n.t('common.messages')}</div>
+      <div />
+    </>
+  )
+}
+
+const ValidationReportRow = ({ user, survey, row, idx, offset }) => {
+  const i18n = useI18n()
+
+  let keysHierarchy = row.keysHierarchy
+  if (keysHierarchy[0].nodeDefUuid === null) {
+    keysHierarchy = keysHierarchy.slice(1)
+  }
+  keysHierarchy = keysHierarchy.concat({ keys: row.keysSelf || {}, nodeDefUuid: row.nodeDefUuid })
+
+  const getKeyValues = keys => Object.values(keys).reduce((values, value) => values.concat(value === null ? '' : value), [])
+
+  const path = keysHierarchy.reduce((p, h) => {
```

Comment:
"path" is not the final path (String) but it should be actually "pathParts".
Maybe it's better to use in this case a R.pipe and join the parts after the reduce... 
In the reduce iterator function, p and h are not so intuitive... p is the accumulator (acc, accPath or something like this) h is the hiearchyPart, isn't it?

Reply:
Renamed as you suggested